benchmark/wildtime_benchmarks/example_pipelines/yearbook.yaml

```diff
@@ -42,9 +42,8 @@ data:
   transformations: []
   bytes_parser_function: |
     import torch
-    import numpy as np
     def bytes_parser_function(data: bytes) -> torch.Tensor:
-      return torch.from_numpy(np.frombuffer(data, dtype=np.float32))
+      return torch.frombuffer(bytearray(data), dtype=torch.float32).reshape(3, 32, 32)
 
```

Comment:
This is to remove a warning. During training this constantly appears:
![image](https://github.com/eth-easl/modyn/assets/100585941/7e94708d-2390-4730-91da-ace5e3e8ad18)

Because the `bytes` itself is not writable and `np.frombuffer` just re-interprets the data as a numpy array, and presents it (as writable data) to pytorch.

The correctness of the modified one is tested by:
```python
import torch
import numpy as np
data = torch.rand(size=(3, 32, 32), dtype=torch.float32)
# save the tensor to a file
with open('test.bin', 'wb') as f:
    f.write(data.numpy().tobytes())


# read the tensor from the file
with open('test.bin', 'rb') as f:
    data_bytes = f.read()

# the old way which produces a warning
# read_data = torch.from_numpy(np.frombuffer(bytes, dtype=np.float32).reshape(3, 32, 32))
# the new way
read_data = torch.frombuffer(bytearray(data_bytes), dtype=torch.float32).reshape(3, 32, 32)

assert torch.allclose(data, read_data)

```

Reply:
I am a bit confused here, unfortunately. What gets passed is a `memoryview`, not `bytes` (and this PR only changes the type hint, which is wrong now). Because the OnlineDataset will always pass a memoryview. Changing the type hint to bytes here should not actually impact the code, it's just a hint, but in this case (from my understanding) a wrong hint, since the parsing function always gets passed a memoryview.

The reason for the old code is that we want to avoid unnecessary in-memory copies. We will never write to this tensor (so it will not cause undefined behavior). With your new version, we create a new `bytearray`, which copies the data out of the memoryview. That's exactly what the old code tries to avoid. I think a better approach would be a way to supress the log this line causes and use the (more efficient) old code:

```
import warnings
import torch

def bytes_parser_function(data: bytes) -> torch.Tensor:
    with warnings.catch_warnings():
        warnings.simplefilter("ignore", category=UserWarning)
        return torch.frombuffer(data, dtype=torch.float32).reshape(3, 32, 32)
```

Reply:
Sorry, I should change back the type hint.

I was just not sure if the warning really hurts anything or it is okay. But your explanation makes sense! If we are sure what returns from this function is never overwritten, we should keep the old implementation.